src/Service.Tests/Configuration/OpenTelemetryTests.cs

```diff
@@ -0,0 +1,105 @@
+#if NET8_0_OR_GREATER
+using System;
+using System.IO;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OpenTelemetry.Exporter;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+using static Azure.DataApiBuilder.Service.Tests.Configuration.ConfigurationTests;
+
+namespace Azure.DataApiBuilder.Service.Tests.Configuration;
+
+/// <summary>
+/// Contains tests for OpenTelemetry functionality.
+/// </summary>
+[TestClass, TestCategory(TestCategory.MSSQL)]
+public class OpenTelemetryTests
+{
+    public TestContext TestContext { get; set; }
+
+    private const string CONFIG_WITH_TELEMETRY = "dab-open-telemetry-test-config.json";
+    private const string CONFIG_WITHOUT_TELEMETRY = "dab-no-open-telemetry-test-config.json";
+    private static RuntimeConfig _configuration;
+
+    /// <summary>
+    /// Creates runtime config file with specified telemetry options.
+    /// </summary>
+    /// <param name="configFileName">Name of the config file to be created.</param>
+    /// <param name="isTelemetryEnabled">Whether telemetry is enabled or not.</param>
+    /// <param name="telemetryConnectionString">Telemetry connection string.</param>
+    public static void SetUpTelemetryInConfig(string configFileName, bool isOtelEnabled, string otelEndpoint, string otelHeaders, OtlpExportProtocol? otlpExportProtocol)
+    {
+        DataSource dataSource = new(DatabaseType.MSSQL,
+            GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL), Options: null);
+
+        _configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions: new(), restOptions: new());
+
+        TelemetryOptions _testTelemetryOptions = new(OpenTelemetry: new OpenTelemetryOptions(isOtelEnabled, otelEndpoint, otelHeaders, otlpExportProtocol, "TestServiceName"));
+        _configuration = _configuration with { Runtime = _configuration.Runtime with { Telemetry = _testTelemetryOptions } };
+
+        File.WriteAllText(configFileName, _configuration.ToJson());
+    }
+
+    /// <summary>
+    /// Cleans up the test environment by deleting the runtime config with telemetry options.
+    /// </summary>
+    [TestCleanup]
+    public void CleanUpTelemetryConfig()
+    {
+        File.Delete(CONFIG_WITH_TELEMETRY);
+        File.Delete(CONFIG_WITHOUT_TELEMETRY);
+        Startup.OpenTelemetryOptions = new();
+    }
+
+    /// <summary>
+    /// Tests if the services are correctly enabled for Open Telemetry.
+    /// </summary>
+    [TestMethod]
+    public void TestOpenTelemetryServicesEnabled()
```

Comment:
can you add some comments on how this test is able to verify that openTelemetry is enabled?